src/codec2_fifo.c

```diff
@@ -82,15 +82,17 @@ int codec2_fifo_write(struct FIFO *fifo, short data[], int n) {
     }
     else {
 
-	/* This could be made more efficient with block copies
-	   using memcpy */
-
 	pdata = data;
-	for(i=0; i<n; i++) {
-	    *pin++ = *pdata++;
-	    if (pin == (fifo->buf + fifo->nshort))
-		pin = fifo->buf;
-	}
+        if ((pin + n) >= (fifo->buf + fifo->nshort))
```

Comment:
Not necessary for test_fifo to pass, but since the TBD was already in the comments, I figured I might as well. 👍